lib/fuel-rest.js

```diff
@@ -106,6 +106,7 @@ FuelRest.prototype.apiRequest = function(options, callback) {
 
 					// check if we should retry req
 					if(helpers.isValid401(res) && retry) {
+						self.AuthClient.invalidateToken();
```

Comment:
We should really just skip the entire retry on auth failure if `options.headers.Authorization` was provided, since it wouldn't be using the managed token.